File: javascript/src/Creational/Prototype/AbstractSpoon.js
(function() {
  define([], function() {
    var AbstractSpoon;
    AbstractSpoon = (function() {
      function AbstractSpoon() {
        console.warn("in AbstractSpoon ctor");
        throw "Can't instantiate abstract class, AbstractSpoon";
      }

      AbstractSpoon.prototype.spoonName = null;

      AbstractSpoon.prototype.setSpoonName = function(spoonName) {
        return this.spoonName = spoonName;
      };

      AbstractSpoon.prototype.getSpoonName = function() {
        return this.spoonName;
      };

      AbstractSpoon.prototype.clone = function() {
        return Object.create(this);
      };

      return AbstractSpoon;

    })();
    return AbstractSpoon;
  });

}).call(this);

File: csharp/src/Creational/Prototype/AbstractSpoon.cs
namespace Creational.Prototype {


//AbstractSpoon.java - One of Two Prototypes

public abstract class AbstractSpoon {
   protected string spoonName; 
    
   public void setSpoonName(string spoonName) {
       this.spoonName = spoonName;
   }
   public string getSpoonName() {
       return this.spoonName;
   }
   
   public object clone() {
       object obj = null;
       obj = this.clone();
       return obj;
   }
}

}
